### ISSUE TITLE
Support jsonpath to pipe data between batch request

### ISSUE THREAD

**Issue by chipsenkbeil** [chipsenkbeil/distant#73]
In some way, it would be handy to support piping the result of one step of a batch request to another if we run them sequentially. Currently, we spawn a new task per request data, so we would need an explicit flag or some way to infer that we need to run a subset of request data sequentially. A dependency graph would highlight which tasks need to wait on others. Alternatively, for any request that has a dependency, we create a oneshot per dependency have the spawned task wait on all oneshots.

https://crates.io/crates/jsonpath_lib provides the best API to do this where we would serialize the response into json, serialize the request into json, iterate over each value within the request, and replace its value with the result of a jsonpath selector on the previous response.

Or something in that manner. I'm not satisified with the above strategy, but having a way to plug in previous values is helpful in cases where we want to read a file or directory and thereby would do a metadata call, check the file type to be file or dir, and perform a different request based on the result.
